File: 프로그래머스/0/181895. 배열 만들기 3/배열 만들기 3.cs
using System;
using System.Collections.Generic;

public class Solution 
{
    public int[] solution(int[] arr, int[,] intervals) 
    {
        List<int> arrList = new List<int>();
        int row = intervals.GetLength(0);
        
        for (int r=0; r<row; r++) {
            for (int c=intervals[r,0]; c<=intervals[r,1]; c++) {
                arrList.Add(arr[c]);
            }
        }
        
        int[] answer = arrList.ToArray();
        return answer;
    }
}

File: 프로그래머스/0/181895. 배열 만들기 3/배열 만들기 3.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> arr, vector<vector<int>> intervals) 
{
    vector<int> ans;
    for(auto v : intervals)
    {
        for(int i=v[0]; i<=v[1]; i++)
        {
            ans.push_back(arr[i]);
        }
    }
    return ans;
}